### ISSUE TITLE
Support building and linking C/C++ code to Rust `oak_loader`

### ISSUE THREAD

**Issue by tiziano88** [project-oak/oak#724]
We will need to keep supporting implementing some of the runtime in C++, and therefore:

- call from Rust to C++ (e.g. implement some part of a pseudo-node in C++ that can be invoked from Rust)
- call from C++ to Rust (e.g. the C++ implementation of a pseudo-node may need to read and write to channels)

In order to build the C++ code, we will probably keep relying on Bazel, and invoke the relevant command from a Rust `build.rs` script.

Ref.:

- #723 
- https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorustc-link-libkindname
- https://doc.rust-lang.org/rustc/command-line-arguments.html#option-l-link-lib

**Comment by tiziano88** [comment]
@daviddrysdale assigning to you to figure out what is actually necessary, and how to go about it. Perhaps it would be useful to target a specific piece of functionality that we expect to be keeping in C++, e.g. the [roughtime client](https://github.com/project-oak/oak/tree/master/oak/server/time).

cc @conradgrobler

**Comment by tiziano88** [comment]
In fact I think roughtime is particularly nice in terms of doing this incrementally, since it may be encapsulated as a stateless C++ function that gets invoked from Rust when necessary to retrieve the time (if we are fine with re-establishing the network connection every time, as opposed to keeping it around as part of a local state).

**Comment by daviddrysdale** [comment]
No longer relevant after #1016 &ndash; we now have a pure Cargo build for the server-side code.
